src/db/VectorDB.cpp

```diff
@@ -0,0 +1,77 @@
+#include <mw/db/VectorDB.h>
+#include "common/Database.h"
+#include "common/Serializable.h"
+
+VectorDB::VectorDB(const std::string& name, libmw::IDBWrapper* pDBWrapper, libmw::IDBBatch* pBatch)
+    : m_name("V" + name), m_pDatabase(std::make_unique<Database>(pDBWrapper, pBatch))
```

Comment:
The database will be a little more compact if you just define a prefix character like we had for UTXOs ('U').